### ISSUE TITLE
[homekit] "No response" error on all openHAB items

### ISSUE THREAD

**Issue by radokristof** [openhab/openhab2-addons#3522]
Dear Community and Developers!

I have been using the homekit addon now for months and I have a problem with it, which I found on the forum, most of us has this problem.
Sometimes the whole Home becomes unavailable, all the items in the Home app shows "No response" message. I have tried to debug it, but I can't find any relevant info, openhab does not show any error or log message when this happens. Also it happens very randomly... Sometimes works for days, sometimes just for hours.
The only solution is to restart openhab (but somehow it works after a while, no need to restart, but this can take hours).

Tell me how can I help to solve this bug!

Ways to reproduce the bug:
Use the builtin HomeKit addon and enable on a device the Hub function.

Solution:
It seems that the Hub feature causes this issue. Temporary fixes:
Turn off the Hub
Use the temporary workaround by @andrey-yantsen 

Thanks!

**Comment by andrey-yantsen** [comment]
Same problem here. I've enabled TRACE-log on the binding couple days ago and those are last log messages related to homekit, before items become unavailable:
```
2018-05-06 00:24:22.839 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Received message of length 0. Existing buffer is 0
2018-05-06 00:24:22.840 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Returning 0 results
2018-05-06 00:24:22.840 [INFO ] [.hap.impl.http.impl.AccessoryHandler] - Terminated homekit connection from /192.168.88.67:49425
2018-05-06 00:25:12.016 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Received message of length 0. Existing buffer is 0
2018-05-06 00:25:12.017 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Returning 0 results
2018-05-06 00:25:12.017 [INFO ] [.hap.impl.http.impl.AccessoryHandler] - Terminated homekit connection from /192.168.88.68:52146
2018-05-06 00:27:34.333 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Received message of length 0. Existing buffer is 0
2018-05-06 00:27:34.333 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Returning 0 results
2018-05-06 00:27:34.334 [INFO ] [.hap.impl.http.impl.AccessoryHandler] - Terminated homekit connection from /192.168.88.66:52228
```

After this events those 3 clients can't communicate with HomeKit (unfortunately I don't have any more HomeKit clients available), but, unfortunately, I'm unable to confirm if communication dies always after those messages. `192.168.88.67` is my AppleTV with HomeKit hub.

BTW, @rkrisi. it's not required to restart entire OpenHab instance, for me it's enough to just to refresh homekit bundle via karaf console, just send following command:
```
bundle:refresh org.openhab.io.homekit
```
And after few seconds the items will return.

P.S. I'm using OpenHab 2.2-stable under Docker, on amd64 virtualbox VM, but the issue is annoys me since the time when I've used OpenHab 2.1-stable on Raspberry Pi 3.

**Comment by andrey-yantsen** [comment]
As for now I've created a [workaround using python and mqtt](https://gist.github.com/andrey-yantsen/0359bb4a74df60b73f6038ae2d87c9fc) to monitor amount of active homekit connections and refresh homekit bundle when last connection lost.

**Comment by wborn** [comment]
Thanks adding a bounty @andrey-yantsen! :smile:

**Comment by rkrisi** [comment]
@andrey-yantsen Thanks, I'll try your method!
Anyway, how can I change the logging level in openHab?

Now I don't know what about anyone's Home app, but sometimes it just shows that connected items not responding, but usually the openHab thing (or I don't know what it is called in Home app) and all other items are not responding...

**Comment by andrey-yantsen** [comment]
> Anyway, how can I change the logging level in openHab?
For my workaround it's enough to use default INFO log level, but if you really want to change loglevel to TRACE for homekit bundle — just say `log:set TRACE com.beowulfe.hap` in your karaf console.

**Comment by piejanssens** [comment]
Ping @beowulfe

**Comment by rkrisi** [comment]
My original post on the forum:
[https://community.openhab.org/t/homekit-no-response-on-all-openhab-items/43540](url)

**Comment by rkrisi** [comment]
I have tried logging and as @andrey-yantsen said, there is no real error message before the connection dies.
It seems that it stops working after all the connections are lost to HomeKit devices. But why? Also we have 2 iPads which is connected to the HomeKit and always at home (one of them is the Hub), so I don't know why it sometimes loses connection and why does it stops working after losing connection.

**Comment by piejanssens** [comment]
This seems to be related to the fact that you are using a hub in your configuration. You could confirm this by removing the hub from the equation for a while and see if the issue still occurs.

**Comment by andrey-yantsen** [comment]
I'm not really sure if using a hub is the root if this issue: I'm also using a hub (Apple TV), but I can see in the logs that my phone connects directly to openhab instance. 
BTW, everything work correctly for almost 2 weeks now, even without my workaround.

**Comment by rkrisi** [comment]
@piejanssens I don't think that Hub is the issue, since if you are at home, it connects directly to openHab... However now I'll try it without enabled.
@andrey-yantsen Yes, I wanted to say the same until yesterday... it worked very well, without your workaround for a lot of days (haven't counted, but I noticed that it was a long time ago since I restarted the service). I don't know why this happens...

**Comment by rkrisi** [comment]
Dear Everyone!

I have tried out using HomeKit with turned off Hub, and it didn't produced the "No Response" error. However I can't say that this is the cause of the error, because I had to restart the whole openHab several times due to that my smarthome is only in "beta", so lots of changes happens these times...

Secondly, 2 days ago I have updated openHab to the stable 2.3.0 release and I haven't experienced this issue (I have turned the Hub back on) and also some other issues were gone for me with this update.
I will continue testing it and report it if I found out anything

**Comment by aihysp** [comment]
hi please tell me if the update did the trick...
I will update also, for now I will stick with 2.2

**Comment by rkrisi** [comment]
For me, 2.3 solved a lot of issue (and seems that this HomeKit error also)... However I saw on the forum that usually most people suffer from upgrading to 2.3 (think they already have a more complex OH setup than mine...)

**Comment by rkrisi** [comment]
No it didn't solve it completely... It worked for much-much longer time than with 2.2, but now it is stuck again at "No response".

**Comment by andrey-yantsen** [comment]
I’ve disabled the hub a couple days ago, and everything is fine for now. Except for items which randomly loose rooms binding, but it may be happining only after items-file updates.

**Comment by rkrisi** [comment]
@andrey-yantsen Thanks, I'll try again with Hub disabled... see if it is really changes anything
Are you on 2.3 already?

**Comment by andrey-yantsen** [comment]
Yep

-- 
Andrey Yantsen
Hotellook.com developer
Tel.: +7-999-218-0373
Skype: andrey_janzen

**Comment by andrey-yantsen** [comment]
I've re-enabled the hub back, because without it my items keep randomly disappear and lose rooms assignment :\ At list with hub issues I have a workaround.

**Comment by radokristof** [comment]
I still don't have much item exposed to HomeKit (2 item yet...), but I don't have this problem. For me it remembers every setting...
However turning off Hub, solved the "No response" error, but it will be great that one time we could use the Hub feature...

**Comment by andrey-yantsen** [comment]
I have 54 items in HomeKit :)

-- 
Andrey Yantsen
Hotellook.com developer
Tel.: +7-999-218-0373
Skype: andrey_janzen

**Comment by radokristof** [comment]
@andrey-yantsen I just started this openHab and Home Automation thing a few months ago, but I will add a few new things during this summer. I will share my experencies after that :) Hope that HomeKit will receive an update with bugfixes and potentially new things in openHab (personally I really want shutters)!

**Comment by codefalk** [comment]
Just increased the bounty

**Comment by jscottnz** [comment]
I have this problem too. I do not have a hub. I am on 2.2.

(Un)fortunately I have a 2 week vacation but I can look into this when I get back.

**Comment by radokristof** [comment]
@jscottnz first I think try updating to 2.3.0. As I said earlier, it is much better for me. As I remember on 2.2.0 I also tested without Hub and had the same problem as you. On 2.3.0 if you turn off Hub, it will work... however see @andrey-yantsen what he said, that his configuration loses the room assigments when it is turned off... I dont have this problem since I have a few items. Test you conf and post your results here!

**Comment by andrey-yantsen** [comment]
As for now the HomeKit works for 2 weeks with Hub enabled (apple tv). About the second issue, where items randomly loose room assignment, — it's not really so random and it's not related to Hub, it's just a new error introduces with 2.3.0, see #3687.

**Comment by jj74** [comment]
I have the same issue. After using Homkit with OH 2.1 on an old Raspi 2, i`ve changed to an Raspi 3 (with an external SSD). First it worked fine for me, but suddenly homekit stopped working. It seems that it`s happening, when i disconnect (ant reconnect to) from my wifi. Then I have to restart a several times and sometimes the connectivity comes back again, but sometimes not. Then i have to delete the bridge and rebind it again, but that does not work anymore. Homekit on iOS says "couldn`t connect to openHAB".

**Comment by Dan1001** [comment]
I have also failed to find a pattern in when homekit stops working. I thought it was linked to item updates, so I put all the homekit items in a separate items file, which I didn't touch. That didn't help - the homekit binding breaks every day or so anyway.

**Comment by piejanssens** [comment]
Possibly related to some of you: https://github.com/openhab/openhab2-addons/issues/1822

**Comment by jj74** [comment]
Here`s an part of the error message from the log:
2018-07-11 08:32:36.192 [ERROR] [ulfe.hap.impl.pairing.PairingManager] - Exception encountered while processing pairing request
com.nimbusds.srp6.SRP6Exception: Bad client credentials
	at com.beowulfe.hap.impl.pairing.HomekitSRP6ServerSession.step2(HomekitSRP6ServerSession.java:331) [270:org.openhab.io.homekit:2.3.0]
	at com.beowulfe.hap.impl.pairing.SrpHandler.step2(SrpHandler.java:76) [270:org.openhab.io.homekit:2.3.0]
	at com.beowulfe.hap.impl.pairing.SrpHandler.handle(SrpHandler.java:47) [270:org.openhab.io.homekit:2.3.0]
	at com.beowulfe.hap.impl.pairing.PairingManager.handle(PairingManager.java:44) [270:org.openhab.io.homekit:2.3.0]
	at com.beowulfe.hap.impl.connections.HttpSession.handlePairSetup(HttpSession.java:107) [270:org.openhab.io.homekit:2.3.0]
	at com.beowulfe.hap.impl.connections.HttpSession.handleRequest(HttpSession.java:51) [270:org.openhab.io.homekit:2.3.0]
	at com.beowulfe.hap.impl.connections.ConnectionImpl.doHandleRequest(ConnectionImpl.java:51) [270:org.openhab.io.homekit:2.3.0]
	at com.beowulfe.hap.impl.connections.ConnectionImpl.handleRequest(ConnectionImpl.java:46) [270:org.openhab.io.homekit:2.3.0]
	at com.beowulfe.hap.impl.http.impl.AccessoryHandler.channelRead0(AccessoryHandler.java:47) [270:org.openhab.io.homekit:2.3.0]
	at com.beowulfe.hap.impl.http.impl.AccessoryHandler.channelRead0(AccessoryHandler.java:15) [270:org.openhab.io.homekit:2.3.0]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105) [270:org.openhab.io.homekit:2.3.0]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308) [270:org.openhab.io.homekit:2.3.0]
	at io.netty.channel.AbstractChannelHandlerContext.access$600(AbstractChannelHandlerContext.java:32) [270:org.openhab.io.homekit:2.3.0]
	at io.netty.channel.AbstractChannelHandlerContext$7.run(AbstractChannelHandlerContext.java:299) [270:org.openhab.io.homekit:2.3.0]
	at io.netty.util.concurrent.DefaultEventExecutor.run(DefaultEventExecutor.java:36) [270:org.openhab.io.homekit:2.3.0]
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112) [270:org.openhab.io.homekit:2.3.0]
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137) [270:org.openhab.io.homekit:2.3.0]
	at java.lang.Thread.run(Thread.java:748) [?:?]
2018-07-11 08:32:36.196 [INFO ] [ap.impl.http.HomekitClientConnection] - 401 /pair-setup
2018-07-11 08:32:36.198 [TRACE] [pl.http.impl.NettyHomekitHttpService] - WRITE UnpooledUnsafeDirectByteBuf(ridx: 0, widx: 72, cap: 256) [/192.168.1.150:57901]:
00000000 48 54 54 50 2F 31 2E 31 20 34 30 31 20 55 6E 61 HTTP/1.1 401 Una
00000010 75 74 68 6F 72 69 7A 65 64 0D 0A 43 6F 6E 74 65 uthorized..Conte
00000020 6E 74 2D 4C 65 6E 67 74 68 3A 20 30 0D 0A 43 6F nt-Length: 0..Co
00000030 6E 6E 65 63 74 69 6F 6E 3A 20 6B 65 65 70 2D 61 nnection: keep-a
00000040 6C 69 76 65 0D 0A 0D 0A                         live....
2018-07-11 08:32:36.242 [INFO ] [.hap.impl.http.impl.AccessoryHandler] - Terminated homekit connection from /192.168.1.150:57901

Still not working/pairing after reboot.

Jesco

**Comment by andrey-yantsen** [comment]
@jj74 if it's not working after reboot — it's another problem.

**Comment by jj74** [comment]
@andrey-yantsen is it? I think it`s exactly the same issue, as discribed above. I`m also using an ATV4 as a hub. When I restart the Pi, the Homekit App cannot connect to OpnHAB (but shows it).

**Comment by andrey-yantsen** [comment]
@jj74 yep. Me and Kristof experience slightly different problem: Homekit works after restart, and no errors are visible in the logs. It would be quite strange if with the same problem and the same OpenHab version you suddenly would be able to get some error in logs, related to the same problem :)

**Comment by radokristof** [comment]
I haven't encountered an error in the previous weeks, but Hub is disabled for me now.
@andrey-yantsen Interesting, but if I use openHab normally, restart it sometimes because of other things, my config didn't lose room associations (Now I have more switches than before, around 8-9). However if I refresh the binding with the command, it did lose every item and adds it to the 'Living Room'

**Comment by aihysp** [comment]
not sure if the issue is completely resolved, 
but for the last two weeks its been cool 

sometimes items are NA but most of the time connected and even after a complete reboot of 
OH machine 

also, I am not connected to the hub(Ipad Air 2) anymore
i will connect it now and see if there is any difference

**Comment by radokristof** [comment]
@aihysp same for me, it is working fine for me now without hub, but if I enable the hub it stops working after a few hours

**Comment by AndrewMorsillo** [comment]
Same issue here. Homekit works well for awhile then stops until the binding is refreshed.

**Comment by aihysp** [comment]
@radokristof i have tested with Hub on for the last two weeks and even with OH reboots i dont get disconnected, on really rare times i see items as NA but most of the time working great!

**Comment by Scorched81** [comment]
I'm also experiencing this issue of connection loss between openhab and my HomeKit hub (ATV4K). After reading here and enabling trace logging, all was good for days. Until tonight, when suddenly my connection was gone and all items showed no response. There was nothing out of the ordinary in the logs (other than what I've seen passing by for days, except for the following 👍 

2018-09-11 21:40:39.726 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Received message of length 0. Existing buffer is 0
2018-09-11 21:40:39.735 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Returning 0 results
2018-09-11 21:40:39.741 [INFO ] [.hap.impl.http.impl.AccessoryHandler] - Terminated homekit connection from /10.0.0.30:60375
2018-09-11 21:41:23.094 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Received message of length 0. Existing buffer is 0
2018-09-11 21:41:23.098 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Returning 0 results
2018-09-11 21:41:23.101 [INFO ] [.hap.impl.http.impl.AccessoryHandler] - Terminated homekit connection from /10.0.0.32:50714
2018-09-11 21:41:30.747 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Received message of length 0. Existing buffer is 0
2018-09-11 21:41:30.752 [TRACE] [ons.LengthPrefixedByteArrayProcessor] - Returning 0 results
2018-09-11 21:41:30.756 [INFO ] [.hap.impl.http.impl.AccessoryHandler] - Terminated homekit connection from /10.0.0.40:49440
2018-09-11 21:41:48.279 [ERROR] [lfe.hap.impl.http.impl.BinaryHandler] - Exception in binary handler

Just to clarify, I'm really a beginner with this, so maybe it can help to investigate this problem further.

**Comment by radokristof** [comment]
@aihysp Thanks for your contribution! What do you mean by NA? Are they seen as 'No Response'? I had few time the same thing (as I can remember I have mentioned here earlier) before, that I didn't had to refresh or restart openHab, it just came back after a few hours (maybe more than that...).

@Scorched81 Yes, I think you have the same issue as we... I don't have an ATV, but the device doesn't really seem to matter here. And yes, the restart is the best option I think. You can refresh just the binding (so you don't have to restart openHab) if you can login to the Karaf console, but for me it loses all the room association, so I have to set it back, one by one...

To be honest, I had bigger problems with my openHab setup in the last weeks, so I didn't had time to investigate this problem better... I have updated my iPad to iOS 12 a few weeks ago, I will try Hub with it. Now I'm using it without Hub and it works without a problem for me.

I'm on OH 2.4 Milestone build now

**Comment by Scorched81** [comment]
@radokristof I just restart it with 'sudo systemctl restart openhab2.service' from command line in an SSH terminal. Never had it lose my room config though.

**Comment by radokristof** [comment]
@Scorched81 If my experiences right and you always restart it that way, you won't lose the room config

**Comment by Scorched81** [comment]
Then I'll continue to do so ;). One thing came to mind, my troubles just recently started when I replaced my Pi 3B for a Pi 3B+ with POE HAT. All was running fine, until I noticed the fan was not spinning. After searching for the cause, it cam clear that my OS was not up to date and had to be updated in order for the fan to be controlled. It was from this point forward that the troubles began. The previous config was running for over 80 days without a hitch. Maybe there is an incompatibility between the code of the OS update and the HomeKit binding ? (just thinking out loud though)

**Comment by radokristof** [comment]
I don’t think so. Are you using openHabian? I have used multiple configs, on multiple platforms and the problem is there. Most people here is using a server (usually x86 based PC) who has a bigger and more complex setup. I’m running mine on a RPi 3B (openHabian) and I use this RPi as well for my media center (Kodi). I don’t have much things in my setup, but it handles well (and I don’t even have active cooling on it as you)

**Comment by Scorched81** [comment]
Agreed, if it occurs on more than one platform, it's not an OS issue. I'm indeed running openhabian and have a mixed setup with HomeKit for light across the house (20something) as well as a thermostat. Also running other bindings as nest binding for nest protects and MQTT to control some Tasmota based Sonoff switches. So not terribly complex. I do however have multiple Pi's running their own dedicated tasks. The active cooling is more for the POE HAT, rather than cooling the CPU due to high loads.

**Comment by aihysp** [comment]
Hi @radokristof yes ,no response, 
what i mean is that before, when i reboot OH, i will need to do the paring again 
now after many reboots it pairs by itself when system boots

only issues i have is from time to time items have no response but its only for a few secs
and then back

**Comment by Scorched81** [comment]
Same here, upgraded to 2.4.0 last weekend and havent had any problems except for a lot of "no route to host / Connection timed out" messages in the log. Now let's hope it stays this way ;)

**Comment by radokristof** [comment]
@Scorched81 If I'm right, that is not really an 'Error'. It would be better if this is treated as a warn by the binding (maybe this is a design bug), but that only shows that on of the HomeKit devices disconnected.

**Comment by Julian0o** [comment]
Have the same Problem. MY Homekit stops working when i change some configuration in OH. It disappears for a while (sometimes days) and then the items in Homekit are back but all items are in the Stanradroom and i have to assign them again.

**Comment by radokristof** [comment]
@Julian0o Try updating to 2.4.0 Milestone release. Since then, I don't have problems with HomeKit and HomeKitHub

**Comment by radokristof** [comment]
It has worked for about 2 weeks in a row, without an issue, but today I got again the No response error on 2.4 M4
However bundle:refresh helps and the room associations are stored

**Comment by radokristof** [comment]
Closing this now, since it seems that the problem is solved!

**Comment by maslyankov** [comment]
Still happening... 👎

**Comment by marbon87** [comment]
Same here

**Comment by Scorched81** [comment]
haven't had this problem since the early versions of 2.4. Have ran into other limits, where you can't use more than 25 accessoires per scene, but that seems Homekit related and not OpenHAB.

**Comment by ccutrer** [comment]
I found the problem! If you have a large number of accessories (I have nearly 150, and that's even after I pruned what I was publishing to HomeKit), HomeKit can send requests that have to be broken up into multiple encryption frames, and the HAP-Java library had a bug in it. Basically it would stop responding to the HomeKit connection, and it would take a while before that connection was dropped a nd a new one established. https://github.com/beowulfe/HAP-Java/pull/64 fixes that bug. I'm also looking into if there is a hard limit of 150 accessories per bridge in HomeKit. The official HAP specifications says a bridge shouldn't expose more than 100. I ran into problems last night at ~150 that I need to see if still happen with this first fix in place. If this is the case, I'll work on the HomeKit OpenHAB add-on to automatically publish multiple bridges if necessary.

**Comment by ccutrer** [comment]
yup, 150 is a hard limit. HomeKit refuses to connect to a bridge that has more than 150 accessories (counting the bridge device itself).

**Comment by beowulfe** [comment]
Since the spec says 100 is the limit, we should probably enforce that in this library and throw an exception when you add the 101st

**Comment by maslyankov** [comment]
I don’t have more than 150 devices/accessories, not now, not when I got the error. The reason was that the homekit configuration in openHAB didn’t have the ip config set.

**Comment by ccutrer** [comment]
@maslyankov : indeed, that is definitely a very common issue. The fix I'm referring to here is rarer, and requires having lots of accessories. But I've also looked into the "have to specify my IP" issue. Turns out HAP-Java doesn't necessarily advertise via Bonjour the same place it's going to listen for requests on. I've also attempted to address that issue with https://github.com/ccutrer/HAP-Java/commit/c8a4c399cfbbf3bc12e08a2774d9aa97ca607536, which seems to be working well on my development machine (a Mac, with like 10 network interfaces of various sorts, where it happened to be binding to a hybrid ipv4/ipv6 address), but I'm not confident enough in it yet to send it upstream.

@beowulfe : probably a good idea. but not until we can have a solution for the OpenHAB HomeKit add-on to allow multiple bridges, otherwise I'll be broken! (I'm at 131 at the moment). I'm still trying to decide if it would be better to have the user specify via config how many bridge instances they want to publish, and then assign accessories via hashing, or try and keep track of which items end up on which bridge, automatically adding bridges as you hit the limit. They both have usability problems. But I'm thinking accessories "jumping" among bridges (and having to reconfigure them in HomeKit - favorite, room, status, etc.) is worth then having to occasionally go pair another bridge.

**Comment by beowulfe** [comment]
I think I'd favor not doing it automatically at all - that will just be confusing. We could use the metadata attributes to allow the user to specify a bridge number. 

It does add configuration overhead, but a user with more than 100 accessories is likely a power user.

**Comment by ccutrer** [comment]
> but a user with more than 100 accessories is likely a power user.

Well, I wasn't a month ago, but I guess I am now... (My HomeKit from OpenHAB has fans, lights, smoke detectors, CO detectors, leak sensors, motion sensors, shades, all properly exposed...)

But yeah, I guess this is a good idea. It would allow me to logically group my bridges - a shades bridge, an alarm items bridge, etc.

**Comment by aihysp** [comment]
I did not have the error for a long time!!!

but about one month ago my Ipad Hub stooped working, 
both my wife iPhone and my iPhone and my apple watch are all working but my hub stooped

**Comment by aaronk6** [comment]
Hi, I have this issue again after it has been working great for a long time. I think it broke after rebooting my server. I didn’t update openHAB though. I’m on 2.5.0.M1 and it has been working fine for weeks.

I tried:

```
bundle:refresh org.openhab.io.homekit
```

(Note that in 2.5.0, it was apparently renamed to `org.openhab.io.homekit`—instead of `com.beowulfe.hap`).

I see in the log that it refreshes all accessories, but even after minutes, the accessories are still marked with “No Response” in the Home app.

I had the `networkInterface` set to the machine’s DNS name, which I suspected could be the problem. However, it’s also not working when a use the machine’s IPv4 address.

I only have 26 devices, so I’m definitely not running into the limit @ccutrer and @beowulfe mentioned.

Any idea what I can do?

**Comment by ccutrer** [comment]
the limit I mentioned is an absolute hard limit. if you have over 150 accessories (including the bridge itself), the home app will refuse to connect to the bridge at all (and the one time I did it with a bridge I was already associated with, it removed all my accessories from the home app, not just show them as Not Responding). 26 accessories may be few enough to hit the packetization problem. I really just depends on how the packets got chopped up over the network. I would highly recommend running the version of HomeKit from either https://github.com/hap-java/openhab2-addons/releases/tag/v2.4.0-homekit-6 or https://github.com/ccutrer/openhab2-addons/releases/tag/2.5.0-cody2 until we can get all of our fixes merged into openhab proper.

**Comment by aaronk6** [comment]
Now using your release from https://github.com/ccutrer/openhab2-addons/releases/tag/2.5.0-cody2. It solved the issue! Thanks so much!

**Comment by malfurion** [comment]
Hi, I’m having the same issue after updating my openhabian.

my openHAB was on version 2.0 or 2.1 before and now I’m on 2.4

HomeKit is working for some days and after some time it’s saying no response
Until waiting some time it fixes itself 

I have only 15 items so would not be a problem with the maximum 

Is there a other known reason why this could happen?

**Comment by kandersen01** [comment]
Why is this thread marked as closed, when its not solved?

**Comment by abbe79** [comment]
Same problem here - openHAB 2.5
Please reopen this issue.

**Comment by mcjimbob** [comment]
I have the same problem. It’s great when I works (When seems to involve clearPairings, restart and re-adding the hub from HomeKit), after a while (a couple of hours usually) all devices go back to “No Response”

Updated to 2.5.4 last night but no change.

Only have 9 items shared from OpenHAB. Apple kit we have is three Apple TV’s (2 of which were recent additions), two iPhones and 2 iPads.

I’ve recently added homebridge but these issues pre-date that addition.

**Comment by frank-f** [comment]
I seem to have the same issue. It's been happening since 2.4.0 on and off. I've been successfully using one of the forked libs for quite a while (months, if I recall correctly), but had to upgrade, when iOS 13 came out.

**Comment by kdub454** [comment]
Fwiw, I had this issue as well - I was able to resolve it by toggling the HomeKit settings on my Apple TV on and off. My issues appeared to be mostly remote-access based.

**Comment by frank-f** [comment]
> Fwiw, I had this issue as well - I was able to resolve it by toggling the HomeKit settings on my Apple TV on and off. My issues appeared to be mostly remote-access based.

I have two HomePods in my setup. I can try to remove those later, but that can't be my solution. I *want* those to be in the setup to be able to use voice commands. This used to work for months, too - it just got worse and worse again with the recent updates of both, openHAB and iOS.

**Comment by 3dr14n** [comment]
> 
> 
> I’m now using your release from https://github.com/ccutrer/openhab2-addons/releases/tag/2.5.0-cody2. It solved the issue! Thanks so much!



> 
> 
> the limit I mentioned is an absolute hard limit. if you have over 150 accessories (including the bridge itself), the home app will refuse to connect to the bridge at all (and the one time I did it with a bridge I was already associated with, it removed all my accessories from the home app, not just show them as Not Responding). 26 accessories may be few enough to hit the packetization problem. I really just depends on how the packets got chopped up over the network. I would highly recommend running the version of HomeKit from either https://github.com/hap-java/openhab2-addons/releases/tag/v2.4.0-homekit-6 or https://github.com/ccutrer/openhab2-addons/releases/tag/2.5.0-cody2 until we can get all of our fixes merged into openhab proper.

This made me move forward from Homekit addon problems after a fresh installation of openHabian. It actually made me pair my iPad Air (iOS 12.4.7) to Openhab. The latest Homekit addon version 2.5.7 snapshot only allows the initialization of Homekit in openHab, pairing was never successful because of `Terminated HomeKit connection from /192.168.10.143:49167` which was the iPad during paring.

**Comment by vlas** [comment]
OH latest stable v3 — same issue. My setup with about 20-30 items suddenly just stopped working with the same problem: no response from all accessories... im not using hubs i.e. apple tv etc, iphones are connected directly to OH...

**Comment by arins** [comment]
I had the same issue but after disabling ipv6 which I read in the forum it worked for me. I followed this guide to disable ipv6
https://techdocs.broadcom.com/us/en/ca-enterprise-software/it-operations-management/network-flow-analysis/21-2/installing/system-recommendations-and-requirements/linux-servers/disable-ipv6-networking-on-linux-servers.html
